src/app/api/comments/route.ts

```
@@ -0,0 +1,146 @@
+        userId: commentData.userId,
+        storyId: commentData.storyId,
+      },
+    });
+
+    return new NextResponse(
+      JSON.stringify({
+        message: "Comment created successfully",
+        commentData: createdComment,
+      }),
+      {
+        status: 200,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  } catch (error) {
+    console.error("Error processing request:", error);
+    return new NextResponse(
+      JSON.stringify({ error: "Internal Server Error" }),
+      {
+        status: 500,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  }
+}
+
+// export async function GET(req: NextRequest) {
+//   try {
+//     const url = new URL(req.url);
+//     const storyId = url.searchParams.get("storyId");
+
+//     if (!storyId) {
+//       return new NextResponse(
+//         JSON.stringify({ error: "storyId is required" }),
+//         {
+//           status: 400,
+//           headers: { "Content-Type": "application/json" },
+//         },
+//       );
+//     }
+
+//     const comments = await prisma.comment.findMany({
+//       where: { storyId: storyId },
+//       include: {
+//         replies: true, // Including nested replies in the response
+//       },
+//     });
+
+//     return new NextResponse(JSON.stringify({ comments }), {
+//       status: 200,
+//       headers: { "Content-Type": "application/json" },
+//     });
+//   } catch (error) {
+//     console.error("Error processing request:", error);
+//     return new NextResponse(
+//       JSON.stringify({ error: "Internal Server Error" }),
+//       {
+//         status: 500,
+//         headers: { "Content-Type": "application/json" },
+//       },
+//     );
+//   }
+// }
+
+export async function GET(req: NextRequest) {
+  try {
+    const url = new URL(req.url);
+    const storyId = url.searchParams.get("storyId");
+    const userId = url.searchParams.get("userId");
```

can you return an error 400 if there is no story id and no user id?

---

solved